### PR TITLE
Add more details about setting up OAuth application

### PR DESCRIPTION
**Pull request by Xiaoyang-Gu**
Signed-off-by: guxiaoyang_s20_307 <guxiaoyang@bupt.edu.cn>

### Description :
Update the README about OAuth setup instructions for Casbin Dashboard. 
[reference.](https://github.com/casbin/casbin-forum#installation) 
[Refer to this comment.](https://github.com/casbin/casbin-dashboard/pull/129#issuecomment-729960393)

### Please ensure Checkmark the following :

- [x] - Tested Localy
- [x] - Checked in IE11 and Firefox
- [ ] - Checked in mobile
- [ ] - Backend work
- [ ] - Frontend work
- [ ] - Bug Fixes
- [x] - Enhancements

### CODE REVIEW

**Comment by 438561537** [comment]
@hsluoyz Please review.

File: README.md [hunk 1]
```diff
@@ -28,16 +28,22 @@ go get github.com/casbin/casbin-dashboard
 
 Casbin-dashboard will store its metadata in a MySQL database named: `casbin_metadata`, will create it if not existed. The DB connection string can be specified at: https://github.com/casbin/casbin-dashboard/blob/master/conf/app.conf
 
+- Setup casbin-dashboard to enable some third-party login platform
+
+Casbin-forum provide a way to sign up using Github account, so you may have to get your own `GithubAuthClientID`, `GithubAuthClientSecret` first.
+
+You could get them by clicking on this url: https://github.com/settings/developers , You should set `Homepage URL` to fit your own domain address, for local testing, set`http://localhost:3000`. And set the Authorization callback URL, the same domain address as `Homepage URL`, for local testing, set`http://localhost:3000`.
```
Comment (com-wushuang) [inline README.md:35]: the callback URL should be `http://localhost:3000/callback/signup`

Comment (438561537) [reply README.md:35]: The reason why I set `http://localhost:3000` to be the callback URL for locally test is that I have made the callback Url in
[setting.js](https://github.com/casbin/casbin-dashboard/blob/235323eefd785cd85d8422567e45abd3bd1f5703/web/src/Setting.js#L41).

Comment (com-wushuang) [reply README.md:35]: In the Setting.js file, the `redirect_uri`  in query parameter is  format as`${ClientUrl}/callback/${method}` . In that case, the request call for GitHub Auth Api will generate a query parameter like is `localhost:3000/callback/signup` if the domain is set as `localhost`

 But according to the GitHub doc , [reference](https://docs.github.com/en/free-pro-team@latest/developers/apps/troubleshooting-authorization-request-errors#redirect-uri-mismatch)

> If you provide a `redirect_uri` that doesn't match what you've registered with your application, GitHub will redirect to the registered callback URL with the following parameters summarizing the error
>
> ```
> http://your-application.com/callback?error=access_denied
>   &error_description=The+user+has+denied+your+application+access.
>   &error_uri=/apps/building-integrations/setting-up-and-registering-oauth-apps/troubleshooting-authorization-request-errors/%23access-denied
>   &state=xyz
> ```

Comment (438561537) [reply README.md:35]: Using `http://localhost:3000` to be the callback URL will not lead to mismatch problem. The redirect URL's host and port must exactly match the callback URL and must reference a `subdirectory` of the callback URL.
[reference](https://docs.github.com/en/free-pro-team@latest/developers/apps/authorizing-oauth-apps#redirect-urls)

So both `http://localhost:3000` and `http://localhost:3000/callback/signup` can be used here, but I mentioned if the `method` changed, using `http://localhost:3000/callback/signup` as callback URL may lead to mismatch problem. [Setting.js](https://github.com/casbin/casbin-dashboard/blob/235323eefd785cd85d8422567e45abd3bd1f5703/web/src/Setting.js#L41).

Comment (pranav1698) [reply README.md:35]: Screenshot from 2021-01-05 00-09-10
even after setting the configurations for the **Homepage URL** and **Callback URL** to `http://localhost:3000`, I am getting the above error 
@Xiaoyang-Gu can you help?

Comment (hsluoyz) [reply README.md:35]: @Xiaoyang-Gu

Comment (Xiaoyang-Gu) [reply README.md:35]: @pranav1698 I recommend you to fix the authentication header. [link](https://github.com/casbin/casbin-dashboard/pull/136)

Comment (pranav1698) [reply README.md:35]: @Xiaoyang-Gu can you please tell me where and what changes needs to be done in the master branch?
Still, can't figure out what is the reason for this error?

Comment (Xiaoyang-Gu) [reply README.md:35]: @pranav1698 Can you show me your console log or other error message? I cannot get what error you meet.

Comment (pranav1698) [reply README.md:35]: @Xiaoyang-Gu changed the port to 3030 and it worked
Screenshot from 2021-01-06 23-03-16

Comment (Xiaoyang-Gu) [reply README.md:35]: OK

**Pull request merged**